nc/source/component/Transform.cpp

```diff
@@ -128,82 +150,167 @@ namespace nc
     void Transform::Set(const Vector3& pos, const Quaternion& quat, const Vector3& scale)
     {
         IF_THROW(HasAnyZeroElement(scale), "Transform::Set - Invalid scale(elements cannot be 0)");
-        m_matrix = ToScaleMatrix(scale) * ToRotMatrix(quat) * ToTransMatrix(pos);
+        m_localMatrix = ToScaleMatrix(scale) * ToRotMatrix(quat) * ToTransMatrix(pos);
```

Comment:
In the essence of DRY - create a private function that takes in a local matrix and sets it, then propagates it via calling UpdateWorldMatrix(); and UpdateChildren().

eg:

SetAndApplyTransform(XMMatrix localMatrix)
{
    m_localMatrix = std::move(localMatrix);
    UpdateWorldMatrix();
    UpdateChildren();
}

Then, in each of these setters you can just call SetAndApplyTransform(<some operation that generates the local transform>)